pkg/kube/client/client.go

```diff
@@ -120,6 +144,21 @@ func configForProvider(ctx context.Context, local client.Client, providerConfigN
 					return nil, errors.Wrap(err, errInjectAzureCredentials)
 				}
 			}
+		case kconfig.IdentityTypeUpboundTokens:
+			switch id.Source { //nolint:exhaustive
+			case xpv1.CredentialsSourceInjectedIdentity:
+				return nil, errors.Errorf("%s is not supported as identity source for identity type %s",
+					xpv1.CredentialsSourceInjectedIdentity, kconfig.IdentityTypeUpboundTokens)
+			default:
+				tkn, err := resource.CommonCredentialExtractor(ctx, id.Source, b.local, id.CommonCredentialSelectors)
+				if err != nil {
+					return nil, errors.Wrap(err, errExtractUpboundCredentials)
+				}
+
+				if err := upbound.WrapRESTConfig(ctx, rc, strings.TrimSpace(string(tkn)), b.store); err != nil {
```

Comment:
if this was a problem before, maybe add a comment that trimspace actually is needed and why